packages/core/react/use-chat.ts

```diff
@@ -159,10 +159,38 @@ const getStreamedResponse = async (
 
   if (isComplexMode) {
     while (true) {
-      const { done, value } = await reader.read()
+      const { done, value: partialValue } = await reader.read()
       if (done) {
         break
       }
+
+      let value = partialValue
+
+      if (value[value.length - 1] !== 10) {
+        // if the last character is not a newline (ASCII code 10), we need to read more chunks
+        let buffer = new Uint8Array(value.length * 2);
```

Comment:
We can store an array of chunk and avoid the reallocations. When we process, count the exact size of all buffered chunks and assign.

Reply:
Yes, you are right! I pushed a new commit to do exactly that!